File: Algorithms/Game Theory/Nimble Game.py
import os

#
# Complete the 'nimbleGame' function below.
#
# The function is expected to return a STRING.
# The function accepts INTEGER_ARRAY s as parameter.
#


def nimbleGame(s):
    xor_sum = 0
    for i, coins in enumerate(s):
        if coins % 2 != 0:
            xor_sum ^= i

    if xor_sum == 0:
        return "Second"
    else:
        return "First"


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    t = int(input().strip())

    for t_itr in range(t):
        n = int(input().strip())

        s = list(map(int, input().rstrip().split()))

        result = nimbleGame(s)

        fptr.write(result + '\n')

    fptr.close()

File: Algorithms/Game Theory/Nimble Game.rs
use std::env;
use std::fs::File;
use std::io::{self, BufRead, Write};

/*
 * Complete the 'nimbleGame' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts INTEGER_ARRAY s as parameter.
 */

fn nimbleGame(s: &[i32]) -> String {
    let xor_sum = s.iter()
        .enumerate()
        .filter(|&(_, &coins)| coins % 2 != 0)
        .fold(0, |acc, (i, _)| acc ^ i);

    if xor_sum == 0 {
        "Second".to_string()
    } else {
        "First".to_string()
    }
}

fn main() {
    let stdin = io::stdin();
    let mut stdin_iterator = stdin.lock().lines();

    let mut fptr = File::create(env::var("OUTPUT_PATH").unwrap()).unwrap();

    let t = stdin_iterator.next().unwrap().unwrap().trim().parse::<i32>().unwrap();

    for _ in 0..t {
        let n = stdin_iterator.next().unwrap().unwrap().trim().parse::<i32>().unwrap();

        let s: Vec<i32> = stdin_iterator.next().unwrap().unwrap()
            .trim_end()
            .split(' ')
            .map(|s| s.to_string().parse::<i32>().unwrap())
            .collect();

        let result = nimbleGame(&s);

        writeln!(&mut fptr, "{}", result).ok();
    }
}
